### DISCUSSION TITLE
Is there any reason that speaks against using mitmproxy as a tls proxy for web filtering in a home network?

### DISCUSSION THREAD

**Discussion by maritain48** [mitmproxy/mitmproxy#7155] [Q&A]
First of all thanks for the great work also the docs! Coming from squid proxy this is a big contrast.

About the suitability of using mitmproxy as a tls proxy for web filtering: E.g. does mitmproxy in its default config violate any standard like http? E.g. squid proxy docs include warnings about configuration options which use violate e.g. http standard:
http://www.squid-cache.org/Doc/config/request_header_access/
In Germany it is generally allowed to use a tls proxy: https://www.bsi.bund.de/SharedDocs/Downloads/DE/BSI/Grundschutz/Kompendium/IT_Grundschutz_Kompendium_Edition2023.pdf?__blob=publicationFile&v=4#download=1

**Comment by mhils** [comment]
> About the suitability of using mitmproxy as a tls proxy for web filtering: E.g. does mitmproxy in its default config violate any standard like http?

I'm no aware of (major) violations. That being said, I don't think it's a great fit. You will likely run into performance issues. At the end of the day, a browser extension is likely more suitable.

**Reply by maritain48** [reply]
Thanks for the reply! If I may ask, which smaller violations are you aware of if any? Regarding performance, the performance achieved while running it locally on my laptop is fine for me. I will still have to see though, how it performs on the raspberrypi 3 I use as router/firewall. Maybe I need to upgrade it. Local solutions like browser extension are more easily bypassed compared to a tls proxy, which is why I want to use it. And sorry a bit ot, are you aware of any mitmproxy systemd unit files e.g. created by any linux distro?
